pkg/operator/pool/controller.go

```diff
@@ -102,12 +102,16 @@ func (c *PoolController) onAdd(obj interface{}) {
 
 func (c *PoolController) onUpdate(oldObj, newObj interface{}) {
 	//oldPool := oldObj.(*Pool)
-	newPool := newObj.(*Pool)
+	pool := newObj.(*Pool)
 
 	// if the pool is modified, allow the pool to be created if it wasn't already
-	err := newPool.create(c.rookClient)
-	if err != nil {
-		logger.Errorf("failed to create (modify) pool %s. %+v", newPool.ObjectMeta.Name, err)
+	if err := pool.create(c.rookClient); err != nil {
+		logger.Errorf("failed to create (modify) pool %s. %+v", pool.ObjectMeta.Name, err)
+	}
+
+	// update pool
+	if err := pool.update(c.rookClient); err != nil {
```

Comment:
This would imply that you can update anything in the pool (although only the size would be updated). Can you validate so that if anything else changes except the size, then we through an error. Perhaps the CRD validation using json schema would apply. Have you look at this?